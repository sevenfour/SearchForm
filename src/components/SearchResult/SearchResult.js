import React from 'react';
import PropTypes from 'prop-types';

import clsx from 'clsx';

import Tile from '../Tile';

import styles from './SearchResult.module.css';

const SearchResult = props => {
  const {
    className,
    name,
    image
  } = props;

  return (
    <Tile className={clsx(styles.container, className)}>
      <figure className={styles.figure}>
        <img
          alt={name}
          className={styles.image}
          src={image}
        />
        <figcaption className={styles.figcaption}>
          {name}
        </figcaption>
      </figure>
    </Tile>
  );
};

SearchResult.propTypes = {
  className: PropTypes.string,
  image: PropTypes.string,
  name: PropTypes.string
};

export default SearchResult;
