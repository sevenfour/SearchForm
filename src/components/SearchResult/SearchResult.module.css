
.container {
  --padding: 20px;
}

.figure {
  margin: 0;
  padding: var(--padding);
}

.image {
  width: 100%;
}

.figcaption {
  padding-top: var(--padding);
}
