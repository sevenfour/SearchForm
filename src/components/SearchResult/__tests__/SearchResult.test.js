import React from 'react';
import { shallow } from 'enzyme';

import SearchResult from '..';

describe('<SearchResult /> component tests', () => {

  it('renders', () => {
    const result = shallow(
      <SearchResult />
    );

    expect(result).toMatchSnapshot();
  });

});
