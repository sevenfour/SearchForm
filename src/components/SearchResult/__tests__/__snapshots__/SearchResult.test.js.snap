// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResult /> component tests renders 1`] = `
<Tile
  className="container"
>
  <figure
    className="figure"
  >
    <img
      className="image"
    />
    <figcaption
      className="figcaption"
    />
  </figure>
</Tile>
`;
